src/main/scala/com/typesafe/sbt/packager/distroless/DistrolessPlugin.scala

```diff
@@ -0,0 +1,262 @@
+package com.typesafe.sbt.packager
+package distroless
+
+import com.typesafe.sbt.SbtNativePackager.Universal
+import com.typesafe.sbt.packager.Keys.{defaultLinuxInstallLocation, maintainer, scriptClasspath}
+import com.typesafe.sbt.packager.archetypes.scripts.*
+import com.typesafe.sbt.packager.docker.DockerPlugin.UnixSeparatorChar
+import com.typesafe.sbt.packager.docker.{Cmd, CmdLike, DockerPlugin, ExecCmd}
+import sbt.*
+import sbt.Keys.{discoveredMainClasses, mainClass, mappings, streams}
+
+/**
+  * ==Distroless Plugin==
+  *
+  * This plugin helps you build docker containers with distroless images
+  *
+  * ==Configuration==
+  *
+  * In order to configure this plugin take a look at the available [[com.typesafe.sbt.packager.docker.DockerKeys]]
+  *
+  * ==Requirements==
+  *
+  * You need docker to have docker installed on your system and be able to execute commands. Check with a single
+  * command:
+  *
+  * {{{
+  * docker version
+  * }}}
+  *
+  * @example
+  *   Enable the plugin in the `build.sbt`
+  *   {{{
+  *    enablePlugins(DistrolessPlugin)
+  *   }}}
+  */
+object DistrolessPlugin extends AutoPlugin {
+
+  override def requires: Plugins = DockerPlugin
+
+  object autoImport extends Keys
+
+  import DistrolessPlugin.autoImport.*
+  import DockerPlugin.autoImport.*
+
+  override def projectConfigurations: Seq[Configuration] = Seq()
+
+  // Some of the default values are now provided in the global setting based on
+  // sbt plugin best practice: https://www.scala-sbt.org/release/docs/Plugins-Best-Practices.html#Provide+default+values+in
+  override lazy val globalSettings: Seq[Setting[?]] =
+    Seq(
+      dockerBaseImage := "gcr.io/distroless/java",
+      dockerEntrypoint := Seq("/usr/bin/java"),
+      distrolessDebuggerPort := None
+    )
+
+  override lazy val projectSettings: Seq[Setting[?]] = Seq(dockerCommands := {
+    val base = dockerBaseImage.value
+    val baseDirectory = (Docker / defaultLinuxInstallLocation).value
+    val dependencies = (Docker / scriptClasspath).value
+    val debugPort = distrolessDebuggerPort.value
+    val compileMainClass = (Compile / mainClass).value
+    val compileDiscoveredMainClasses = (Compile / discoveredMainClasses).value
+    val log = streams.value.log
+
+    val layerMappings = (Docker / dockerLayerMappings).value
+    val layerIdsAscending = layerMappings.map(_.layerId).distinct.sortWith { (a, b) =>
+      // Make the None (unspecified) layer the last layer
+      a.getOrElse(Int.MaxValue) < b.getOrElse(Int.MaxValue)
+    }
+
+    val cmd = if (dockerCmd.value.isEmpty) {
```

Comment:
In my original approach I duplicated `dockerCmd` to be an `Option[Seq[String]]`. This way users could define an empty CMD, or leave alone with a `None`. This in turn allowed this code to know explicitly what the user wanted.

With this preset, even if the user wants an empty CMD, the plugin will still calculate it and include in the Dockerfile. This, I feel, is a decent compromise. With a distroless image the CMD would probably always be defined to at least specify which JAR the user wants to run